ale/transformation.py

```diff
@@ -115,35 +115,45 @@ def from_spice(cls, *args, sensor_frame, target_frame, center_ephemeris_time, ep
         for s, d in time_dependent_frames:
             quats = np.zeros((len(times), 4))
             avs = np.zeros((len(times), 3))
-            for j, time in enumerate(times):
-                state_matrix = spice.sxform(spice.frmnam(s), spice.frmnam(d), time)
-                rotation_matrix, avs[j] = spice.xf2rav(state_matrix)
-                quat_from_rotation = spice.m2q(rotation_matrix)
-                quats[j,:3] = quat_from_rotation[1:]
-                quats[j,3] = quat_from_rotation[0]
-
-            rotation = TimeDependentRotation(quats, times, s, d, av=avs)
-            frame_chain.add_edge(rotation=rotation)
+            try:
+                for j, time in enumerate(times):
+                    state_matrix = spice.sxform(spice.frmnam(s), spice.frmnam(d), time)
+                    rotation_matrix, avs[j] = spice.xf2rav(state_matrix)
+                    quat_from_rotation = spice.m2q(rotation_matrix)
+                    quats[j,:3] = quat_from_rotation[1:]
+                    quats[j,3] = quat_from_rotation[0]
+
+                rotation = TimeDependentRotation(quats, times, s, d, av=avs)
+                frame_chain.add_edge(rotation=rotation)
+            except Exception as e:
+                pass
 
         for s, d in constant_frames:
-            quats = np.zeros(4)
-            rotation_matrix = spice.pxform(spice.frmnam(s), spice.frmnam(d), times[0])
-            quat_from_rotation = spice.m2q(rotation_matrix)
-            quats[:3] = quat_from_rotation[1:]
-            quats[3] = quat_from_rotation[0]
+            try:
+                quats = np.zeros(4)
+                rotation_matrix = spice.pxform(spice.frmnam(s), spice.frmnam(d), times[0])
+                quat_from_rotation = spice.m2q(rotation_matrix)
+                quats[:3] = quat_from_rotation[1:]
+                quats[3] = quat_from_rotation[0]
 
-            rotation = ConstantRotation(quats, s, d)
+                rotation = ConstantRotation(quats, s, d)
 
-            frame_chain.add_edge(rotation=rotation)
+                frame_chain.add_edge(rotation=rotation)
+            except Exception as e:
+                pass
 
         return frame_chain
 
     @staticmethod
-    def frame_trace(reference_frame, ephemeris_time):
+    def frame_trace(reference_frame, ephemeris_time, nadir=False):
         frame_codes = [reference_frame]
         _, frame_type, _ = spice.frinfo(frame_codes[-1])
         frame_types = [frame_type]
 
+        if nadir:
+            frame_codes.append(1)
+            frame_types.append(1)
```

Comment:
Why is this appending two? It should only be appending a single instance of J2000

Reply:
Which list should get the value? frame codes or frame types?

Reply:
oh I completely missed this was two lists x.x